docs/docs/setup-vast/tune.md

```diff
@@ -0,0 +1,220 @@
+---
+sidebar_position: 5
+---
+# Tune
+
+This section describes tuning knobs that have a notable effect on system
+performance.
+
+## Batching: Table Slices
+
+VAST processes events in batches. Because the structured data has the shape of a
+table, we call these batches *table slices*. The following options control their
+shape and behavior.
+
+:::note Implementation Note
+Table slices are implemented as *Record Batches* in Apache Arrow.
+:::
+
+### Size
+
+Most components in VAST operate on table slices, which makes the table slice
+size a fundamental tuning knob on the spectrum of throughput and latency. Small
+table slices allow for shorter processing times, resulting in more scheduler
+context switches and a more balanced workload. But the increased pressure on the
+scheduler comes at the cost of throughput. Conversely, a large table slice size
+creates more work for each actor invocation and makes them yield less frequently
+to the scheduler. As a result, other actors scheduled on the same thread may
+have to wait a little longer.
+
+The option `vast.import.batch-size` sets an upper bound for the number of events
+per table slice.
+
+The option merely controls number of events per table slice, but not
+necessarily the number of events until a component forwards a batch to the next
+stage in a stream. The CAF streaming framework uses a credit-based flow-control
+mechanism to determine buffering of tables slices.
+
+:::tip
+Setting `vast.import.batch-size` to 0 causes the table slice size to be
+unbounded and leaves it to other parameters to determine the actual table slice
+size.
+:::
+
+### Import Timeout
+
+The `vast.import.batch-timeout` option sets a timeout for forwarding buffered
+table slices to the importer. If the timeout fires before a table slice reaches
+`vast.import.batch-size`, then the table slice will contain fewer events and
+ship immediately.
+
+## Memory usage and caching
+
+The amount of memory that a VAST server process is allowed to use can currently
+not be configured directly as a configuration file option. Instead of such a
+direct tuning knob, the memory usage can be influenced through the configuration
+of the caching, catalog and disk monitor features.
+
+### Caching
+
+VAST groups table slices with the same schema in a *partition*. When building a
+partition, the parameter `max-partition-size` sets an upper bound on the number
+of records in a partition, across all table slices.
```

Comment:
This should probably also mention `active-partition-timeout`, which can override this upper bound.

Reply:
Good call.

Reply:
Added a sentence.